it/controllers/benefits/reimbursed/OtherBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,447 @@
+          }
+        }
+      }
+    }
+
+    "redirect to another page when a valid request is made and then" should {
+      "Update the other items question to no, and other benefits to none when the user chooses no, redirects to CYA if prior submission" which {
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.no)
+
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = true, cyaModel(Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+          authoriseAgentOrIndividual(isAgent = false)
+          urlPost(pageUrl(taxYearEOY), body = form, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "redirects to the check your details page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+        }
+
+        "updates other items question to no and other benefits to none" in {
+          lazy val cyaModel = findCyaData(taxYearEOY, employmentId, userRequest).get
+          cyaModel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.otherItemsQuestion)) shouldBe Some(false)
+          cyaModel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.otherItems)) shouldBe None
+        }
+      }
+
+      "Update the other items question to no, and other benefits to none when the user chooses no," +
+        "redirects to next question if not prior submission" which {
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.no)
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = false, cyaModel(Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+          authoriseAgentOrIndividual(isAgent = false)
+          urlPost(pageUrl(taxYearEOY), body = form, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        //          TODO: This will need updating to the next page in the flow when the navigation is hooked up
+        "redirects to the check your details page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+        }
+
+        "updates other items question to other benefits to none" in {
+          lazy val cyaModel = findCyaData(taxYearEOY, employmentId, userRequest).get
+          cyaModel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.otherItemsQuestion)) shouldBe Some(false)
+          cyaModel.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.otherItems)) shouldBe None
+        }
+      }
+
```

Comment:
It might be good to have a test where the user answers yes and it goes to the amount page

Reply:
test added